### PR TITLE
 XIVY-10351 Add multi tasks tab

### PR DESCRIPTION
**Pull request by ivy-lli**
- Add task switch gateway mock data
- Load task data from 'config/task' or 'config/tasks'

### CODE REVIEW

File: packages/editor/src/context/useTaskData.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { TaskDataAccess } from '@axonivy/inscription-protocol';
+import { createContext, useContext } from 'react';
+import { useData } from './useData';
+
+export const TaskDataContextInstance = createContext<string>('config/task');
+
+export function useTaskData<K extends keyof TaskDataAccess>(
+  path: K
+): [TaskDataAccess[K], TaskDataAccess[K], (newData: TaskDataAccess[K]) => void];
+export function useTaskData<T>(path: string): [T, T, (newData: T) => void] {
```
Comment (ivy-rew) [inline packages/editor/src/context/useTaskData.ts:10]: typescript can get complex too :sweat_smile:

File: packages/protocol/src/data/task-data.ts [hunk 2]
```diff
@@ -18,22 +18,27 @@ export interface CustomField {
...
+
 export interface TaskDataAccess {
-  'config/task/name': string;
-  'config/task/description': string;
-  'config/task/category': string;
-  'config/task/priority': string;
+  name: string;
+  description: string;
+  category: string;
+  priority: string;
```
Comment (ivy-rew) [inline packages/protocol/src/data/task-data.ts:30]: looks simple :+1: 
... maybe we should/Could already extract the 'expiry' to sub-interface too ... and keep the selector path a single key.

File: packages/core/src/mock/data/task-data.ts [hunk 3]
```diff
@@ -3,22 +3,31 @@ import { CustomField } from '@axonivy/inscription-protocol';
...
+  };
+}
+
+export interface TasksData {
+  config: {
+    tasks: Task[];
+  };
+}
+
+interface Task {
```
[the remark this answers is not in the document]
Comment (ivy-lli) [reply packages/core/src/mock/data/task-data.ts:16]: Don't get your point... This is only for the mocked data. currently, only the name is set, so I will remove the other stuff...

File: packages/editor/src/context/useTaskData.ts (hunk 1 above)
Comment (ivy-lli) [reply packages/editor/src/context/useTaskData.ts:10]: Yes ;)

File: packages/protocol/src/data/task-data.ts (hunk 2 above)
Comment (ivy-lli) [reply packages/protocol/src/data/task-data.ts:30]: currently not possible, needs to be a string with a "/" as a separator. But I'm also not very happy with this...

**Comment by ivy-lli** [comment]
I merge it for now. I want to do some improvements to the data access anyway later.

**Pull request merged**